tools/src/main/scala/scala/scalanative/build/Filter.scala

```diff
@@ -61,9 +61,14 @@ private[scalanative] object Filter {
       // included files to the link phase
       val includePaths = allPaths.map(_.abs).filter(include)
 
+      val gcFlag = {
+        val gc = config.compilerConfig.gc.toString
+        s"-DSCALANATIVE_GC_${gc.toUpperCase}"
+      }
+
       val projectConfig = config.withCompilerConfig(
         _.withCompileOptions(
-          config.compileOptions ++ gcIncludePaths.map("-I" + _)
+          config.compileOptions ++ gcIncludePaths.map("-I" + _) :+ gcFlag
```

Comment:
If we use C preprocessor conditions we could potentially get rid of the `includes` paths passed here and just compile everything. It would also make sure that is coverts everything.

Reply:
The latest update uses one include path and cleans up the build.

Reply:
I left the `gc` path because we have so many things in the `nativelib`.